### PR TITLE
Feat/#67 resource lifecycle record

### PR DESCRIPTION
**Pull request by asulis**
### Description:
This PR adds am entity to track the change of state in one or more negotiation resources, during the related lifecycle.

### Checklist:

_Make sure you tick all the boxes bellow if they are true or do not apply:_

- [X ] I have performed a self review of my code
- [X ] My code follows [Google Java Code style](https://google.github.io/styleguide/javaguide.html)
- [X] I have made my code as simple as possible
- [X ] I have added unit tests and the code coverage has not decreased
- [X ] I have updated the documentation in all relevant places

### CODE REVIEW

File: src/test/java/eu/bbmri/eric/csit/service/negotiator/integration/service/NegotiationLifecycleServiceImplTest.java [hunk 1]
```diff
@@ -172,6 +175,17 @@ void sendEventForResource_approvedNegotiation_Ok() throws IOException {
         NegotiationResourceState.REPRESENTATIVE_CONTACTED,
         resourceLifecycleService.sendEvent(
             negotiationDTO.getId(), "biobank:1:collection:2", NegotiationResourceEvent.CONTACT));
+    // check that the negotition has a resource record with the assigned state
```
Comment (RadovanTomik) [inline src/test/java/eu/bbmri/eric/csit/service/negotiator/integration/service/NegotiationLifecycleServiceImplTest.java:178]: please remove comment and replace with appropriately named method that experesses the logic in its name

File: src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Negotiation.java [hunk 2]
```diff
@@ -156,4 +177,11 @@ public Set<Resource> getResources() {
         .flatMap(request -> request.getResources().stream())
         .collect(Collectors.toUnmodifiableSet());
   }
+
+  public Resource lookupResource(Set<Resource> resources, String resourceId) {
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Negotiation.java:181]: please refactor. This method is not needed I believe or at least should not be public.

File: src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Negotiation.java [hunk 3]
```diff
@@ -134,6 +144,17 @@ public void setStateForResource(String resourceId, NegotiationResourceState stat
     currentStatePerResource.put(resourceId, state);
   }
 
+  public void addNewRecordToNegotiationResourceStateHistory(
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Negotiation.java:147]: make the part of setStateForressource.

**Review by RadovanTomik — CHANGES_REQUESTED** [review]

File: src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Negotiation.java (hunk 3 above)
Comment (asulis) [reply src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Negotiation.java]: Fixed in 673d2d0e5aa92b06fda147cf34949dd7444d4de4

File: src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Negotiation.java (hunk 2 above)
Comment (asulis) [reply src/main/java/eu/bbmri/eric/csit/service/negotiator/database/model/Negotiation.java]: Fixed in 673d2d0e5aa92b06fda147cf34949dd7444d4de4

File: src/test/java/eu/bbmri/eric/csit/service/negotiator/integration/service/NegotiationLifecycleServiceImplTest.java (hunk 1 above)
Comment (asulis) [reply src/test/java/eu/bbmri/eric/csit/service/negotiator/integration/service/NegotiationLifecycleServiceImplTest.java]: Fixed in 673d2d0e5aa92b06fda147cf34949dd7444d4de4

**Pull request merged**